tracer/src/Datadog.Trace/Debugger/Symbols/SymbolPdbExtractor.cs

```diff
@@ -0,0 +1,250 @@
+
+                            var name = nestedHandle.FullName(MetadataReader);
+                            if (!Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.AsSpan(local.Type).SequenceEqual(
+                                Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.AsSpan(name)))
+                            {
+                                continue;
+                            }
+
+                            var nestedType = MetadataReader.GetTypeDefinition(nestedHandle);
+                            var fields = nestedType.GetFields();
+                            int index = 0;
+                            var added = ArrayPool<string>.Shared.Rent(fields.Count);
+                            foreach (var fieldHandle in fields)
+                            {
+                                if (fieldHandle.IsNil)
+                                {
+                                    continue;
+                                }
+
+                                var field = MetadataReader.GetFieldDefinition(fieldHandle);
+                                if (field.Name.IsNil)
+                                {
+                                    continue;
+                                }
+
+                                var fieldName = MetadataReader.GetString(field.Name);
+                                var fieldNameAsSpan = Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.AsSpan(fieldName);
+                                if (Datadog.Trace.VendoredMicrosoftCode.System.MemoryExtensions.IndexOf(fieldNameAsSpan, generatedClassPrefix, StringComparison.Ordinal) == 0)
+                                {
+                                    continue;
+                                }
+
+                                bool contains = false;
+                                for (int j = 0; j < added.Length; j++)
+                                {
+                                    if (added[j] == fieldName)
+                                    {
+                                        contains = true;
+                                        break;
+                                    }
+                                }
+
+                                if (contains)
+                                {
+                                    continue;
+                                }
+
+                                added[index] = fieldName;
+                                localsSymbol.Add(new Symbol
+                                {
+                                    Name = fieldName,
+                                    Type = field.DecodeSignature(new TypeProvider(false), 0),
+                                    SymbolType = SymbolType.Local,
+                                    Line = local.Line
+                                });
+                                index++;
+                            }
+
+                            ArrayPool<string>.Shared.Return(added, true);
```

Comment:
bad practice: you should try/finally the usage of the rented array. If you fall on exception, the array won't be returned to the pool.

Reply:
Fixed in f74ef445ed4a6220e0849d0be2ba5866e48a3fa5